bdp_app/lib/pages/ChatPage.dart

```diff
@@ -188,7 +275,24 @@ class _ChatPageState extends State<ChatPage> {
                 color: Theme.of(context).cardColor,
               ),
               child: _textComposerWidget(),
-            )
+            ),
+            _emojiInput ? new EmojiPicker(
+              rows: 3,
+              columns: 7,
+              recommendKeywords: ["racing", "horse"],
+              numRecommended: 10,
+              onEmojiSelected: (emoji, category) {
+                //this adds emoji always at the end of the text which is not nice
+                textEditingController.text = textEditingController.text + emoji.emoji;
+
+                /* this code would enable emoji input anywhere in the text, but cursorPos does very strange things therefore
+                not possible
+                var cursorPos = textEditingController.selection;
+                textEditingController.text = textEditingController.text.substring(0, cursorPos.start)
+                    + emoji.emoji + textEditingController.text.substring(cursorPos.start, textEditingController.text.length);
+                */
```

Comment:
Damit sollte das Emoji an die korrekte Position gesetzt werden und der Cursor hinter dem Emoji landen:
```                
var cursorPos = textEditingController.selection.start;
textEditingController.text = textEditingController.text.substring(0, cursorPos)
                    + emoji.emoji + textEditingController.text.substring(cursorPos, textEditingController.text.length);
textEditingController.selection = TextSelection.fromPosition(TextPosition(offset: cursorPos + 2));
setState(() {});
```

Reply:
hmja, also damit kann man emojis auch im text einbauen aaaber der cursor wird trotzdem an den anfang gesetzt. mir scheint das problem hatten schon andere:
https://github.com/flutter/flutter/issues/11416
war mir dann irgendwie zu viel struggle fürs erste. ich hab deinen code jetzt mal reingebastelt dann kann man zumindest mal überall emojis reinbasteln

Reply:
oh we, und wenn der cursor dann am anfang steht dann kann man keine emojis einfügen...

Reply:
naja also da muss ich nochmal ran, aber nicht jetzt jetzt brummt mein schädel :)

Reply:
sooo, habe jetzt einen workaround gefunden. der internet ist so toll <3